edge-util/src/Microsoft.Azure.Devices.Edge.Util/azureLogAnalytics/AzureLogAnalytics.cs

```diff
@@ -0,0 +1,96 @@
+    /* https://dejanstojanovic.net/aspnet/2018/february/send-data-to-azure-log-analytics-from-c-code/
+    */
+
+    public sealed class AzureLogAnalytics
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<AzureLogAnalytics>();
+        static AzureLogAnalytics instance = null;
+
+        AzureLogAnalytics()
+        {
+        }
+
+        public static AzureLogAnalytics GetInstance()
+        {
+            Preconditions.CheckNotNull(instance);
+            return instance;
+        }
+
+        public static AzureLogAnalytics InitInstance()
+        {
+            if (instance == null)
+            {
+                instance = new AzureLogAnalytics();
+            }
+
+            return instance;
+        }
+
+        public async void PostAsync(string workspaceId, string sharedKey, string content, string logType)
+        {
+            Preconditions.CheckNotNull(workspaceId, "Log Analytic workspace ID cannot be empty.");
+            Preconditions.CheckNotNull(sharedKey, "Log Analytic shared key cannot be empty.");
+            Preconditions.CheckNotNull(content, "Log Analytic content cannot be empty.");
+            Preconditions.CheckNotNull(logType, "Log Analytic log type cannot be empty.");
+
+            const string apiVersion = "2016-04-01";
+
+            try
+            {
+                string dateString = DateTime.UtcNow.ToString("r");
+                Uri requestUri = new Uri($"https://{workspaceId}.ods.opinsights.azure.com/api/logs?api-version={apiVersion}");
+                string signature = this.GetSignature("POST", content.Length, "application/json", dateString, "/api/logs", workspaceId, sharedKey);
+
+                var client = new HttpClient();
+                client.DefaultRequestHeaders.Add("Authorization", signature);
+                client.DefaultRequestHeaders.Add("Accept", "application/json");
+                client.DefaultRequestHeaders.Add("Log-Type", logType);
+                client.DefaultRequestHeaders.Add("x-ms-date", dateString);
+
+                var contentMsg = new StringContent(content, Encoding.UTF8);
+                contentMsg.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                Log.LogDebug(
+                    client.DefaultRequestHeaders.ToString() +
+                    contentMsg.Headers +
+                    contentMsg.ReadAsStringAsync().Result);
+
+                var response = await client.PostAsync(requestUri, contentMsg).ConfigureAwait(false);
+                var responseMsg = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                Log.LogDebug(
+                    ((int)response.StatusCode).ToString() + " " +
+                    response.ReasonPhrase + " " +
+                    responseMsg);
+            }
+            catch (Exception e)
+            {
+                Log.LogDebug(e.Message);
```

Comment:
you don't need to Log 2 times; LogDebug will log even log level is defined as Error since Error level is higher than Debug level.